### PR TITLE
feat: UI for multiple operator service selection

### PR DESCRIPTION
**Pull request by SamerIssam**
## Description

This ticket is refactoring how the select multiple operator page functions as now user can see all selected operators on the same page and can easily remove and review selected services for each operator.

Add testing instructions, if required
Start multiple operators journey and go to /multipleOperatorsServiceList  and start select some services

### CODE REVIEW

File: repos/fdbt-site/src/interfaces/index.ts [hunk 1]
```diff
@@ -562,7 +566,7 @@ export interface ServiceWithErrors {
 }
 
 export interface ServiceType {
-    id: number;
+    id?: number;
```
Comment (alainajaffer) [inline repos/fdbt-site/src/interfaces/index.ts:569]: shouldn't id always be compulsory?

Comment (SamerIssam) [reply repos/fdbt-site/src/interfaces/index.ts:570]: its not for this ticket, but this type is used else where so I did not want to remove it completely.

File: repos/fdbt-site/src/pages/api/multipleOperatorsServiceList.ts [hunk 2]
```diff
@@ -1,105 +1,116 @@
...
+            serviceDetails.push(value);
         });
-    });
-    return {
-        selectedServices,
-        nocCode,
+    // {SelectedServices} to {MultiOperatorInfo}
+    const multiOperatorsService: SelectedServiceByNocCode[] = [];
+    // export interface SelectedServiceByNocCode {
+    //     [key: string]: ServiceWithNocCode[];
```
Comment (NitishAroraKPMG) [inline repos/fdbt-site/src/pages/api/multipleOperatorsServiceList.ts:55]: remove commented code

File: repos/fdbt-site/src/pages/api/multipleOperatorsServiceList.ts [hunk 3]
```diff
@@ -1,105 +1,116 @@
...
+    //output
+    // nocCode,
+    // lineName: splitStrings[0],
+    // lineId: splitStrings[1],
+    // serviceCode: splitStrings[2],
+    // startDate: splitStrings[3],
+    // serviceDescription,
+
+    Object.entries(requestBody)
+        .filter((item) => item[0] !== 'OperatorCount')
```
Comment (NitishAroraKPMG) [inline repos/fdbt-site/src/pages/api/multipleOperatorsServiceList.ts:46]: do we need Object.entries?

File: repos/fdbt-site/src/pages/api/multipleOperatorsServiceList.ts [hunk 4]
```diff
@@ -1,105 +1,116 @@
...
-            multiOpDataToReAddToSession = multiOpDataOnSession;
-        }
-        if ((!req.body || Object.keys(req.body).length === 0) && !selectAll) {
+        if (!req.body || Object.keys(req.body).length === 0) {
             updateSessionAttribute(req, MULTIPLE_OPERATORS_SERVICES_ATTRIBUTE, {
-                multiOperatorInfo: multiOpDataToReAddToSession,
+                multiOperatorInfo: [],
                 errors: [{ id: errorId, errorMessage: 'Choose at least one service from the options' }],
             });
-            redirectTo(res, `${redirectUrl}?selectAll=false`);
```
Comment (NitishAroraKPMG) [inline repos/fdbt-site/src/pages/api/multipleOperatorsServiceList.ts:73]: why we are removing this

**Review by NitishAroraKPMG — COMMENTED** [review]
please do the changes

File: repos/fdbt-site/src/pages/api/multipleOperatorsServiceList.ts (hunk 4 above)
Comment (SamerIssam) [reply repos/fdbt-site/src/pages/api/multipleOperatorsServiceList.ts:73]: what is removed is the (?selecteAll=false). no we don't use it anymore.
Each service has its own status weather if its selected or not inside the services list.

**Review by alainajaffer — APPROVED** [review]

**Pull request merged**